src/services/folder.rs

```diff
@@ -25,54 +25,40 @@ use crate::utils;
 pub struct FolderService;
 
 impl FolderService {
-    pub async fn get(
-        ctx: Arc<Context>,
-        id: Uuid,
-        _reference: String,
-        path: Option<String>,
-        recursive: Option<&String>,
-    ) -> Result<Tree> {
+    pub async fn get(ctx: Arc<Context>, id: Uuid, path: Option<String>) -> Result<Vec<File>, ApiError> {
         let playbook = ctx.client.playbooks().get(&id.to_string()).map_err(ApiError::NotFoundPlaybook)?;
         let source = playbook.preface.repository.unwrap();
 
-        let content = ctx
-            .github_client
+        ctx.github_client
             .contents()
-            .find(&utils::repo(&source.repo)?, &path.unwrap_or_default(), &source.branch.unwrap_or_default())
-            .map_err(|e| ApiError::NotFoundContent(e.to_string()))?;
+            .list(&utils::repo(&source.repo)?, &path.unwrap_or_default(), &source.reference().unwrap())
```

Comment:
`&source.reference().unwrap()` You can determine if a reference is null before using it, and if it is null, just return an error.